### ISSUE TITLE
Epic: Instrument "Activation & Retention" Growth Area Telemetry Data in IDE

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#7326]
| Context             | To provide a better Gitpod experience, we need to ensure that users are able to discover all of Gitpod features. This aspect of the Gitpod experience is referred to as "Activation & Retention". The purpose of this epic is to instrument Gitpod with the necessary data points on all major Gitpod features. |
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Owner               | Product - @loujaybee \ Tech - @akosyakov                                                                                                                                                                                                                                                                        |
| Value               | Gitpod are able to make data-informed future product decisions.                                                                                                                                                                                                                                                 |
| Acceptance Criteria | - Instrument all of the key Gitpod features                                                                                                                                                                                        |
| Growth Area         | `Activation & Retention`                                                                                                                                                                                                                                                                                        |
| Persona(s)          | N/A                                                                                                                                                                                                                                                                                                             |
| Hypothesis          | N/A                                                                                                                                                                                                                                                                                                             |
| Measurement         | N/A                                                                                                                                                                                                                                                                                                             |
| In scope            | - Instrumentation of analytics                                                                                                                                                                                                                                                                                  |
| Out of scope        | - Any functional changes to Gitpod                                                                                                                                                                                                                                                                              |
| Complexities        | N/A                                                                                                                                                                                                                                                                                                             |

**Comment by loujaybee** [comment]
The file events mentioned by @jeanp413 [here](https://github.com/gitpod-io/gitpod/issues/6981#issuecomment-994271182) may also be relevant to this epic.

**Comment by loujaybee** [comment]
We need to integrate this into Mixpanel, but I think we're now waiting on @jakobhero to resolve some issues with Mixpanel vs Mode. CC: @phimae

**Comment by loujaybee** [comment]
Whilst we are still waiting on this to be integrated into PLG areas dashboard [[1](https://www.notion.so/gitpod/PLG-Areas-Dashboard-6ff7bba7eec6413587ac5be7c3fde33a)] I'm going to adjust the scope to be only the instrumentation, and not the visualisation so that we can close this issue now, and raise follow up issues for integration with visualisations / dashboards.

**Comment by loujaybee** [comment]
https://github.com/gitpod-io/gitpod/issues/6894 would also be useful, but let's pick that up in future.
